devito/function.py

```diff
@@ -306,12 +306,12 @@ def argument_defaults(self, alias=None):
 
         :param alias: (Optional) name under which to store values.
         """
-        key = alias or self.name
-        args = ArgumentMap({key: self._data_buffer})
+        key = alias or self
+        args = ArgumentMap({key.name: self._data_buffer})
 
         # Collect default dimension arguments from all indices
-        for i, s, o in zip(self.indices, self.shape, self.staggered):
-            args.update(i.argument_defaults(size=s+o))
+        for i, s, o, k in zip(self.indices, self.shape, self.staggered, key.indices):
```

Comment:
so it works this way? you said it wasn't going to be OK this way...

Reply:
Looks like it does for now